code/edgeidentity/src/main/java/com/adobe/marketing/mobile/edge/identity/IdentityItem.java

```diff
@@ -37,12 +37,15 @@ public final class IdentityItem {
 	private final boolean primary;
 
 	/**
-	 * Creates a new {@link IdentityItem}
+	 * Creates a new {@link IdentityItem}.
+	 * An {@code IdentityItem} should not have an empty or null {@code id} value. An {@link IdentityMap}
+	 * will reject {@code IdentityItem}s with null or empty identifiers.
+	 *
 	 *
 	 * @param id                 id for the item; should not be null
 	 * @param authenticatedState {@link AuthenticatedState} for the item; if none is provided {@link AuthenticatedState#AMBIGUOUS} is used as default
 	 * @param primary            primary flag for the item
-	 * @throws IllegalArgumentException if id is null
+	 * @throws IllegalArgumentException if {@code id} is null
 	 */
 	public IdentityItem(
 		@NonNull final String id,
```

Comment:
`@NotBlank`
https://javaee.github.io/javaee-spec/javadocs/javax/validation/constraints/NotBlank.html

Reply:
This is not a standard AndroidX annotation, but instead from Hibernate-Validator in Java EE, so it's not really available to us.